### PR TITLE
Add support for .docx files

### PR DESCRIPTION
**Pull request by gravesm**
This adds support for reading uploaded docx files. A document module is
added with a factory function for creating a document object from a
Django UploadedFile object. The created document object has a words
property which contains a list of words in the document. In addition,
this uses chardet to determine the encoding of plain text files.

### CODE REVIEW

File: hamlet/theses/document.py
Comment (thatandromeda) [inline hamlet/theses/document.py:61]: I take it `docx` handles all encoding issues internally?

File: hamlet/theses/fixtures/thesis.txt
Comment (thatandromeda) [inline hamlet/theses/fixtures/thesis.txt:1]: 🎉 😆

File: hamlet/theses/forms.py
Comment (thatandromeda) [inline hamlet/theses/forms.py:74]: Do you need a file generated from an actual Windows product to test this on? I can easily supply you with an OS X Office file, and I can get a Windows one also.

**Comment by thatandromeda** [comment]
I fired this up and tried to upload a .docx and did not meet with success:

screen shot 2018-02-06 at 12 48 00 pm
screen shot 2018-02-06 at 12 48 11 pm

**Comment by thatandromeda** [comment]
File upload wasn't working for me, but it turned out this was due to incompatibilities between gensim versions - I've pinned the version number in the Pipfile and now it works. I'll open a ticket for the version incompatibility issue (but it will be very low priority; rewriting the gensim code is thinky work and there are more important ways to spend our time).

File: hamlet/theses/document.py
Comment (gravesm) [reply hamlet/theses/document.py:61]: Yeah, by the time we get text out of it, it's already unicode. I will say that if we decide we don't like how the library I'm using (it's the only one I could find) is performing it probably wouldn't be totally onerous to roll our own. Our needs for pulling text out of the document are very minimal.

**Review by thatandromeda — APPROVED** [review]
LGTM but tell me if you want to do anything with respect to the questions above.

File: hamlet/theses/forms.py
Comment (gravesm) [reply hamlet/theses/forms.py:74]: Probably wouldn't hurt, though I'd guess with what we're doing, any differences in implementation of the standard won't have much of an impact. It's no problem to throw a few more files at it during testing, so why not?

**Comment by gravesm** [comment]
I think I'd like to at least look into updating the tests to go through the actual view, based on the document you sent me before merging.

File: hamlet/theses/forms.py
Comment (thatandromeda) [reply hamlet/theses/forms.py:74]: I've sent you a Mac-generated one, and Matt will be sending you a Windows-generated one shortly.

**Comment by thatandromeda** [comment]
Integration tests ftw.

File: docs/developer.md
Comment (gravesm) [inline docs/developer.md]: What if we added a new setting called something like HAMLET_MODEL that specified the path to an alternate model? This would have the advantage of being more flexible in where it lives and what it's called, as well as being more explicit about what's getting used. If the setting doesn't exist we can fall back to the testmodel.

Comment (thatandromeda) [reply docs/developer.md]: Presumed to be the relative path from the project root? Sure, that works. Will do.

**Pull request merged**
